### PR TITLE
SUMMER THAW: THE UNUSED FEATURES 2: Red Phone

### PR DESCRIPTION
**Pull request by Iamgoofball**
Using the Red Phone now allows you to with a short delay send a message to Central Command.
https://tgstation13.org/phpBB/viewtopic.php?f=5&t=3771&start=100#p94291 requested by STICKYMAYHEM
svqe1jr

### CODE REVIEW

**Comment by RemieRichards** [comment]
Stop ruining one of the only portal references we have.
Revert the red phone change dangit.

**Comment by Iamgoofball** [comment]
@RemieRichards when this shit is so obscure you have to go digging through wiki easter eggs to find it, it stops becoming a reference

**Comment by RemieRichards** [comment]
@Iamgoofball no it's just a very well placed one, it's called subtlety, because yes, that's a thing!

**Comment by PKPenguin321** [comment]
iirc emagging drones is a strictly taboo thing for a lot of reasons
1. they're a ghost role that are highly expendable and have low cost.
Because of this, ghosts that want revenge/want some valids will stalk roboticists and if they have an emag they get free antag roles, and if they die the robo can just make more

2. Drones are supposed to be a passive force in the round, not a large active force. they can repair a bomb breach or disposals breach and save the engineering crew some time. they should not be the talk of everyone on the station and be actively hunted like xenomorphs

3. drones themselves are very powerful, moreso than rogue borgs. they have hands that can manipulate ANYTHING, and teeny tiny little sprites that are very difficult to shoot/attack. they can also move over things like tables and racks

:-1: from me, emagging drones would be cool in a perfect world without meta or grudges but wouldn't work very well at all with what we have now

**Comment by Fox-McCloud** [comment]
Uh...the phone is...a portal reference? Huh.

**Comment by Iamgoofball** [comment]
@PKPenguin321

The only law that changes is the one that tells you to build and improve the station. The other 2 laws about staying out of the way of humans are un-changed.

They'll still get their shit slapped by admins if they start walking up to humans and punching them.

**Comment by PKPenguin321** [comment]
it's still a free unlimited antag roll for any valid hungry ghost that can cause murders, albeit indirectly

and drones deconstructing the station would definitely put them as a large threat and make them way too big of a factor in the round

and they're still super small and impossible to kill

**Comment by Iamgoofball** [comment]
It's not an antag role.

You're not going to be causing murders.

Did you glance over the part where I said that the other 2 laws are unchanged?

**Comment by RemieRichards** [comment]
@Fox-McCloud yes.
In the portal universe there is/was a red phone in the entrance hall to GLaDOS's chamber, one employee had to sit there all day ready to call the phone in an emergency, however GLaDOS was able to flood neutrotoxin and lock doors quick enough to make it pointless.

it's a really NICE, SUBTLE portal ref (go away goof)

**Comment by Fox-McCloud** [comment]
That's...pretty obscure, especially considering red phones being a "last ditch"/hotline has been a "thing" for a very long time: https://en.wikipedia.org/wiki/Hotline

**Comment by Iamgoofball** [comment]
@RemieRichards and giving it this functionality ruins the reference how?

**Comment by Kromgar** [comment]
Damn it goof if your gonna do summer thaw make bags of holding capable of carrying corpses... That or foldable bluespace body bags with corpses in them

**Comment by RemieRichards** [comment]
It loses it's connection to the reference material, for a shitty gimmick.
People who need to contact centcom have the communication console, everyone else can kill/hack to get to that console.

**Comment by Iamgoofball** [comment]
@RemieRichards how do you know it was even supposed to be a portal reference

how do we know it isn't that hotline thing that was linked

because i'm 99% certain it's that hotline thing and not a portal reference

**Comment by KorPhaeron** [comment]
I thought it was supposed to be the whitehouse/kremlin phone

Shows what I know

**Comment by Nienhaus** [comment]
It's clearly the phone from Powerpuff girls.
phone-powerpuff-girls-24455695-400-300

**Comment by Lumipharon** [comment]
Emagging drone = one guy with an emag shitting out infinity drones who all go validboner and assfuck the station.

No thanks.

**Comment by AlexanderUlanH** [comment]
Drones are meant to be something you can rely on to be not fucking shit up when you see them in normal play; everything about them's designed that way.  You can pick up a drone and chuck it around if it's fucking with you, but if it's just breaking floor tiles with an RCD, or smashing windows to space, you'll never catch it.  :-1:, this should not be a thing.

**Comment by Iamgoofball** [comment]
# laws 1 and 2 are 
# still 
# in 
# effect

they can't harm beings or interfere with them. They can start completely ripping crap apart but they can't get their "validboner" on.

**Comment by PKPenguin321** [comment]
\>implying breaching the station and deconstructing the only thing keeping humans out of the vacuum of space isn't murder, albeit indirect

**Comment by Niknakflak** [comment]
>I thought it was supposed to be the whitehouse/kremlin phone
>Shows what I know

Same

**Comment by Lumipharon** [comment]
>not interfer with people
>deconstruct the floors from under them

**Comment by Iamgoofball** [comment]
@Lumipharon it's almost as if there's not any areas on the station where there aren't humans usually that you can deconstruct!

**Comment by Fox-McCloud** [comment]
I admit it's a bit anachronistic to tell them to essentially damage an destroy everything but not interfere with people---yes, you can destroy areas uninhabited, but that's still kinda interfering.

I dunno; I have no problem with them getting an "obey your master" lawset, but I"m not sure how well that will go over on TG--especially with a respawn mechanic.

**Comment by Dorsisdwarf** [comment]
Try "The [src] rings..." instead of "ring..."

Also this should send a message to "Central Troubleshooting"

**Comment by SconesC** [comment]
Emagging drones no

Red phone, hilarious and yes

**Comment by PJB3005** [comment]
Wait what, the red phone is a portal reference? from where?

The icon of the courtesy call's achievement?

**Comment by Iamgoofball** [comment]
@Dorsisdwarf the ring... shows to the mob using it. it's to imply that you hear the ring sound that plays while you're waiting on them to pick up.

**Comment by kingofkosmos** [comment]
"Ring..." can be confusing. I know I would probably think it's trying to tell me I should pick up and answer the call.
"The [src] gives a ringback tone..." perhaps? https://en.wikipedia.org/wiki/Ringback_tone

**Comment by Miauw62** [comment]
I disagree with emagging drones. :-1: until you remove that.

**Comment by xxalpha** [comment]
I don't like either change.

**Comment by Dorsisdwarf** [comment]
I despise feedback like << "Calibration engaged" or "ring". visible_message is fine, or if you must <<, at least make it match every other feedback and say what's doing it

**Comment by ZomgPonies** [comment]
BestPhone

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -210,6 +210,56 @@
 	w_class = 2
 	attack_verb = list("called", "rang")
 	hitsound = 'sound/weapons/ring.ogg'
+	var/being_used = 0
+
+/obj/item/weapon/phone/attack_self(mob/user)
+	if(being_used)
```
Comment (optimumtact) [inline code/game/objects/items/weapons/weaponry.dm]: This whole proc is disgusting

Comment (Iamgoofball) [inline code/game/objects/items/weapons/weaponry.dm]: why

Comment (optimumtact) [inline code/game/objects/items/weapons/weaponry.dm]: It's not your fault, it's just long and ugly to me

**Comment by TheVekter** [comment]
@RemieRichards When it's a reference so obscure that even I don't know about it, it's probably going to be lost on 99% of people.

As for the PR itself, :+1:

**Comment by Giacom** [comment]
>@RemieRichards how do you know it was even supposed to be a portal reference
how do we know it isn't that hotline thing that was linked
because i'm 99% certain it's that hotline thing and not a portal reference

Well, as the creator of the feature-filled red phone I will tell you that it is indeed a reference to Portal, as it was originally placed in front of the AI's doors.

I also had fun adding it and seeing people trying to figure out what it did.

:-1: For actually giving it a feature.

**Comment by RemieRichards** [comment]
I TOLD Y'ALL. BUT DIDJA LISTEN? NAWW!

**Comment by Jacquerel** [comment]
unless the sprite is changed it will still be a reference, and I'd rather have a fun phone than a useless reference

**Comment by Niknakflak** [comment]
Giacom comes from beyond the grave to shatter the doubts.  

I agree to not add the red phone change now, or the drone change. So yea, trash this PR.

**Comment by TheVekter** [comment]
Why not add it, aside from >nik hates fun?

**Comment by Ricotez** [comment]
Put one phone in the AI sat foyer, the other one in the RD's office, and have them connect.

**Comment by TheVekter** [comment]
That sounds a bit broken.

**Comment by Niknakflak** [comment]
Actually the reason is more than "Nik hates fun", its. "Lets not ruin a reference by skewing it and making drones an administrative headache" but yea know

That is cool too vekter

**Comment by SconesC** [comment]
Drone thing is retarded, I've never seen an admin or ex-admin advocate for this idea in general (which has been brought up a ton) because evidently only admins understand the headache that drones already are aka all-access ventcrawling gypsies 

Phone is still hilarious though

**Comment by Iamgoofball** [comment]
Drones were removed guys

**Comment by palpatine213** [comment]
Phone being a portal reference was pretty obvious to me, considering it is in the AI upload.

Just keep the original description, and have it go to an emergency hotline, rather than calling a general front desk number. Why the hell would a red phone NOT call an emergency number anyway?
Also AI should totally get a notification so that it can bolt doors, switch on lasers if it's rogue.

**Comment by Iamgoofball** [comment]
I'll make those changes, palpatine. Minus the AI part, that gets a little TOO snowflakey.

**Comment by SconesC** [comment]
ring ring

does this mean we can have phones in the future

**Comment by Niknakflak** [comment]
Ring ring

This is a space station with a station wide radio, what the fuck do you need phones for

**Comment by SconesC** [comment]
ring ring

suck my dick

**Comment by Niknakflak** [comment]
Ring ring

Go fuck yourself

**Comment by kevinz000** [comment]
Does this mean I can call in ert as Rd in rev?

**Comment by samis** [comment]
Perhaps make the phone lead to the 'AI Technical Support' division? Might make more sense than the Emergency Hotline being located near the AI for some reason?

**Comment by whodaloo** [comment]
fuck yes, the phone in the upload makes you call tech support
and they put you on hold forever

**Comment by RemieRichards** [comment]
@Iamgoofball if you're going to keep being a tool and ruin the reference, ATLEAST remove the "peggy/sue/receptionist" bullhonky, it's not even needed, the two lines around it are fine:
"Nanotrasen emergency hotline, what's your emergency?"

**Comment by Razharas** [comment]
@Cheridan your call on this

File: code/game/objects/items/weapons/weaponry.dm [hunk 2]
```diff
@@ -210,6 +210,56 @@
...
+	user << "[src] rings..."
+	playsound(src, 'sound/weapons/ring.ogg', 40, 1)
+	sleep(20)
+	user << "[src] rings..."
+	playsound(src, 'sound/weapons/ring.ogg', 40, 1)
+	sleep(20)
+	user << "[src] rings..."
+	playsound(src, 'sound/weapons/ring.ogg', 40, 1)
+	sleep(20)
+	say("Central Command Headquarters, this is [pick("Captain", "Ensign", "Lieutenant", "Commander", "Rear Admiral", "Vice Admiral", "Admiral", "Fleet Admiral")] [pick("Bob","Jebediah","Bill")] speaking. What is the situation?")
```
Comment (Dorsisdwarf) [inline code/game/objects/items/weapons/weaponry.dm]: ehehehehehheee

Comment (Jalleo) [inline code/game/objects/items/weapons/weaponry.dm]: That is probably better in a while loop

Comment (Iamgoofball) [inline code/game/objects/items/weapons/weaponry.dm]: What?

**Comment by Aranclanos** [comment]
can i close this i hate it
specially all those sleeps holy shiiiiiiiiiit

**Pull request closed without merging**